### DISCUSSION TITLE
Shared field for blocks? It's wasting bytes in the graphql query

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#2566] [Feature Requests & Ideas]
I have a blocks field with multiple sections with a shared sectionStyle field. Now it can only be fetched with duplicate queries like this: 

![image](https://user-images.githubusercontent.com/5801015/234477733-dea70b28-0bef-453f-8dd5-0d1c5ec22f53.png)

I want sth like this to work:

![image](https://user-images.githubusercontent.com/5801015/234479268-65e96469-9179-4cd4-9819-5c869be92d41.png)

**Comment by Stupidism** [comment]
Anybody facing similar issue like me?

**Reply by Stupidism** [reply]
The possible solution would be allow customizing shared type fields like __typename, blockName and blockType.

**Reply by itzaks** [reply]
Yes this is one of my biggest concerns when using graphql with payload. I've had to repeat all fields.

**Reply by jmikrut** [reply]
We are actively building a solution for this! 

https://github.com/payloadcms/payload/discussions/1410

It's in progress as I write this comment.

**Reply by Stupidism** [reply]
@jmikrut Thanks, the link you shared describes my question much better than myself.

**Reply by Stupidism** [reply]
@jmikrut Sorry, I found this is actually a different feature request after I just tested the new feature released in 1.9.

**Comment by Stupidism** [comment]
I cannot use string templates or gql inline fragments feature because we're using apollo to generate typescript based on the literal graphql query.

**Comment by Stupidism** [comment]
Close for duplication: https://github.com/payloadcms/payload/discussions/1410

**Comment by Stupidism** [comment]
@jmikrut What I'm requesting is a graphql client query feature to saving graphql query bytes on client size, 

<img width="1108" alt="image" src="https://github.com/payloadcms/payload/assets/5801015/eed0d60e-53d3-4517-b9df-ea3ff89e7226">


but not a typescript optimization to saving generated code in code repo.

<img width="1271" alt="image" src="https://github.com/payloadcms/payload/assets/5801015/d9e1db46-f321-46d9-9339-99b1a7b8b655">
